integration-testing/src/main/proto/io/grpc/testing/integration/test.proto

```diff
@@ -58,6 +58,11 @@ service TestService {
   rpc StreamingInputCall(stream StreamingInputCallRequest)
       returns (StreamingInputCallResponse);
 
+  // Start uploading only after receiving a confirmation message from the server.
+  // The server returns the aggregated size of client payload as the final result.
+  rpc CheckedUploadCall(stream StreamingInputCallRequest)
```

Comment:
So what I'm delaying on is that I don't think we want to issue a call like we are here.

Instead, I'm thinking we could have a method like StreamingInputCall in which the server sends headers as soon as the call starts. That could be achieved with an interceptor I believe. I question if we even need a different method and whether instead StreamingInputCall could always respond with headers eagerly.

Since this proto is used by interop testing, I'm trying to prevent duplicating other methods as much as possible. I'm trying to have each method orthogonal and useful for a range of different tests.